### PR TITLE
Pass compilation args using the official way

### PR DESCRIPTION
**Pull request by Bencodes**
Pulling this portion of the Kotlin 1.5.30 support out of the original PR. We should be able to merge this independently so that people can use Kotlin 1.5.30 without rules_kotlin having it as the default.

Original PR here: https://github.com/bazelbuild/rules_kotlin/pull/588

### CODE REVIEW

File: src/main/kotlin/io/bazel/kotlin/builder/tasks/jvm/CompilationArgs.kt [hunk 1]
```diff
@@ -172,28 +169,18 @@ class CompilationArgs(
...
+        flag(flag, transform(option, it))
       }
     }
-
-    oos.flush()
-    flag(flag, transform(Base64.getEncoder().encodeToString(os.toByteArray())))
     return this
   }
+
+  fun list(): List<String> = args.toList()
```
Comment (cgruber) [inline src/main/kotlin/io/bazel/kotlin/builder/tasks/jvm/CompilationArgs.kt:185]: Probably should stick with the toList() naming pattern.

**Review by cgruber — COMMENTED** [review]
This seems fine to me, but I want @restingbull to confirm it's not going to clobber or interfere with his other efforts around 1.5.

File: src/main/kotlin/io/bazel/kotlin/builder/tasks/jvm/CompilationArgs.kt (hunk 1 above)
Comment (restingbull) [reply src/main/kotlin/io/bazel/kotlin/builder/tasks/jvm/CompilationArgs.kt:185]: Can't just merge it as we needed the base64 encoding for older versions of kotlinc. Currently working on a way to preserve versionsing without making too much of a mess.

Comment (restingbull) [reply src/main/kotlin/io/bazel/kotlin/builder/tasks/jvm/CompilationArgs.kt:192]: Added branch to cover pre-1.5

Not the way I would like it, but I don't want to hold up progress while I yak shave with one hand.

**Pull request merged**